pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -174,144 +156,136 @@ public void runSegmentMetrics() {
         }
-        if (nReplicas == 0) {
-          if (nOffline < MaxOfflineSegmentsToLog) {
-            LOGGER.warn("Segment {} of table {} has no online replicas", partitionName, tableName);
-          }
-          nOffline++;
+        if (nReplicasExternal == -1) {
+          nReplicasExternal = (nReplicasIdealMax == 0) ? 1 : 0;
         }
-        nReplicasExternal = ((nReplicasExternal > nReplicas) || (nReplicasExternal == -1)) ? nReplicas : nReplicasExternal;
-      }
-      if (nReplicasExternal == -1) {
-        nReplicasExternal = (nReplicasIdealMax == 0) ? 1 : 0;
-      }
-      // Synchronization provided by Controller Gauge to make sure that only one thread updates the gauge
-      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.NUMBER_OF_REPLICAS,
-          nReplicasExternal);
-      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_REPLICAS,
-          (nReplicasIdealMax > 0) ? (nReplicasExternal * 100 / nReplicasIdealMax) : 100);
-      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.SEGMENTS_IN_ERROR_STATE,
-          nErrors);
-      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_SEGMENTS_AVAILABLE,
-          (nSegments > 0) ? (100 - (nOffline * 100 / nSegments)) : 100);
-      if (nOffline > 0) {
-        LOGGER.warn("Table {} has {} segments with no online replicas", tableName, nOffline);
-      }
-      if (nReplicasExternal < nReplicasIdealMax) {
-        LOGGER.warn("Table {} has {} replicas, below replication threshold :{}", tableName,
-            nReplicasExternal, nReplicasIdealMax);
+        // Synchronization provided by Controller Gauge to make sure that only one thread updates the gauge
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.NUMBER_OF_REPLICAS, nReplicasExternal);
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_REPLICAS,
+            (nReplicasIdealMax > 0) ? (nReplicasExternal * 100 / nReplicasIdealMax) : 100);
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.SEGMENTS_IN_ERROR_STATE, nErrors);
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_SEGMENTS_AVAILABLE,
+            (nSegments > 0) ? (100 - (nOffline * 100 / nSegments)) : 100);
+        if (nOffline > 0) {
+          LOGGER.warn("Table {} has {} segments with no online replicas", tableName, nOffline);
+        }
+        if (nReplicasExternal < nReplicasIdealMax) {
+          LOGGER.warn("Table {} has {} replicas, below replication threshold :{}", tableName, nReplicasExternal,
+              nReplicasIdealMax);
+        }
+      } catch (Exception e) {
+        LOGGER.warn("Caught exception while updating segment status for table {}", e, tableName);
+
+        // Remove the metric for this table
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.NUMBER_OF_REPLICAS, Long.MIN_VALUE);
```

Comment:
Perhaps setting the value to a higher number than Long.MIN_VALUE may help? Otherwise it may be hard to eye-ball a graph showing the values and decipher whether the value was 0, 1, or 2 before the curve takes a plunge. May be -1 will be ok here. Your call.